File: koopey.web/src/app/models/competition.ts
import { Base } from "./base/base";
import { User } from "../models/user";
import { Game } from "../models/game";

export class Competition extends Base {
  public userId!: String;
  public user: User = new User();
  public gameId!: String;
  public game: Game = new Game();
}

File: koopey.android/app/src/main/java/com/koopey/model/Competition.java
package com.koopey.model;

import com.koopey.model.base.Base;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.SuperBuilder;

@Data
@EqualsAndHashCode(callSuper = true)
@SuperBuilder
public class Competition extends Base {
    private String gameId;
    private String userId;
}
